homeassistant/components/shelly/coordinator.py

```diff
@@ -771,7 +772,7 @@ async def shutdown(self) -> None:
             except InvalidAuthError:
                 self.entry.async_start_reauth(self.hass)
                 return
-            except DeviceConnectionError as err:
+            except (DeviceConnectionError, ClientConnectionResetError) as err:
```

Comment:
I think it would be better to catch this upstream if possible and keep it within the `DeviceConnectionError` as it might be relevant in other places we have a disconnection to to this error